### ISSUE TITLE
Consumer creation exceptions are not handled

### ISSUE THREAD

**Issue by KannarFr** [sksamuel/pulsar4s#262]
When creating to consume a non-exisiting namespace, the java pulsar client prints a lot of exceptions like:

```
org.apache.pulsar.client.api.PulsarClientException$LookupException:  Policies not found for <tenant>/<namespace> namespace
```

This is normal, but pulsar4s should provide consumer creation in a Try. WDYT?

**Comment by gmethvin** [comment]
That probably makes sense, but I'd want to make sure we're consistent about using either Try to encapsulate errors anywhere we'd throw. In other words if we make such a change I'd rather break the API all at once rather than changing methods individually to return Try.

That said I'm not really that worried about this case because most people should be using the async APIs anyway.

**Comment by KannarFr** [comment]
@sksamuel WDYTN

**Comment by sksamuel** [comment]
I agree with Greg. We should be either all in on Try or just use exceptions.
Really I should have called these methods dosomethingUnsafe and then dosomething() returns a Try. That doesn't help us now though.

**Comment by KannarFr** [comment]
I agree too. So, let's make breaking changes?

**Comment by sksamuel** [comment]
I don't think we should break all the methods now was kind of what I was meaning unless it was a 3.0 release, but then we need pulsar to do that too.

I'd be happy to add new methods / deprecate old ones.
